packages/runtime/src/runtime.ts

```diff
@@ -400,6 +401,23 @@ export class Runtime {
       txnParameters = txnParams;
     }
     for (const txnParam of txnParameters) {
+      // check if transaction is signed by logic signature,
+      // if yes verify signature and run logic
+      if (txnParam.sign === SignType.LogicSignature) {
+        if (txnParam.lsig === undefined) {
+          throw new TealError(ERRORS.TEAL.LOGIC_SIGNATURE_NOT_FOUND);
+        }
```

Comment:
What happens when user passes sign type as secrect key (instead of logic signature), and passes a logic signature with the txParams? Is it covered or should we add that check?

Reply:
i don't think we can cover that. because this is clearly user's mistake, there is 50% chance that it is correct(and 50% it is wrong).

Reply:
I wanted to know how the transaction is executed if the user makes that mistake (passing secretkey as sign type along with a logic signature). Because in algob i think `signTx` covers that case

Reply:
what if he is right ? and mistakenly sent lsig

Reply:
sending secret key was right, but lsig was mistake or otherwise. this is 50-50

Reply:
also, if secret key doesn't matches with lsig address, logic signature validation will throw error.

Reply:
> what if he is right ? and mistakenly sent lsig

How can the user be right? Is there a transaction where user can send a secret key (knowingly), and passes lsig?
If the user is passing secret key knowingly, then the transaction must be other than stateless asc execution (algoTransfer, assetTransfer, stateful call, ..). Am i missing something?

Reply:
yes, what if he makes a mistake in passing `lsig` , it should not be passed but he passed it anyway.
how will you decide what was the mistake ? passing extra `lsig` or passing wrong `sign`.

Reply:
think about the first case, user passed `secret key` with `lsig` -> this case is clearly confusing for interpreter.
but the question is how will you decide the mistake of user ?

correct case may be `logicSignature` with `lsig` 
or
`SecretKey` with no `lsig`

50-50
does that makes sense ?

Reply:
we can add a warning though, that these should not be passed simultaneously

Reply:
It's still not clear to me how the tx is executed if we pass sk with lsig. Please add a warning/note atleast. I will test it on my end later.

Reply:
ok